met-web/src/components/engagement/form/EngagementFormTabs/AddTeamMemberModal.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useContext, useState } from 'react';
+import Modal from '@mui/material/Modal';
+import { Autocomplete, Grid, Paper, Stack, TextField } from '@mui/material';
+import { MetHeader3, MetLabel, modalStyle, PrimaryButton, SecondaryButton } from 'components/common';
+import { User, USER_GROUP } from 'models/user';
+import { useForm, FormProvider, SubmitHandler, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { addUserToGroup } from 'services/userService/api';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { useAppDispatch } from 'hooks';
+import { EngagementTabsContext } from './EngagementTabsContext';
+import { ActionContext } from '../ActionContext';
+import { openNotificationModal } from 'services/notificationModalService/notificationModalSlice';
+
+const schema = yup
+    .object({
+        user: yup.object().required('A user must be selected'),
+    })
+    .required();
+
+type AddUserForm = yup.TypeOf<typeof schema>;
+
+export const AddTeamMemberModal = () => {
+    const dispatch = useAppDispatch();
+    const { addTeamMemberOpen, setAddTeamMemberOpen, users } = useContext(EngagementTabsContext);
+    const { savedEngagement } = useContext(ActionContext);
+    const [isAdding, setIsAdding] = useState(false);
+
+    const methods = useForm<AddUserForm>({
+        resolver: yupResolver(schema),
+    });
+
+    const {
+        handleSubmit,
+        control,
+        reset,
+        formState: { errors },
+    } = methods;
+
+    const { user: userErrors } = errors;
+
+    const handleClose = () => {
+        setAddTeamMemberOpen(false);
+        reset({});
+    };
+
+    const onSubmit: SubmitHandler<AddUserForm> = async (data: AddUserForm) => {
+        try {
+            setIsAdding(true);
```

Comment:
we should find a better pattern for setIsAdding() kind of things ;its just everywhere 
may be a new ticket

Reply:
Okay, let me know of the task description and I will write a ticket for it